sqlglot/dialects/spark2.py

```diff
@@ -132,6 +133,9 @@ class Parser(Hive.Parser):
             ),
             "STRING": _parse_as_cast("string"),
             "TIMESTAMP": _parse_as_cast("timestamp"),
+            "TO_TIMESTAMP": lambda args: _parse_as_cast("timestamp")(args)
+            if len(args) == 1
+            else format_time_lambda(exp.StrToTime, "spark")(args),
```

Comment:
is this always true? can't you call to_timestamp on a date object?

Reply:
Based on their docs, you can't:

<img width="714" alt="Screenshot 2023-07-18 at 9 23 41 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/3c43cf54-5c72-4b45-8901-b9acd14331e1">

Based on spark-sql's behaviour, you can:

```python
spark-sql (default)> select to_timestamp(cast('2020-01-01' as date));
2020-01-01 00:00:00
Time taken: 4.96 seconds, Fetched 1 row(s)
```

I just found out myself too, assumed the docs would be accurate :)

What do you think we should do about this?